visual-python/src/saucelabs_visual/client.py

```diff
@@ -0,0 +1,185 @@
+from os import environ
+from typing import List
+
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport, BasicAuth
+
+from saucelabs_visual.regions import Region
+from saucelabs_visual.typing import IgnoreRegion, FullPageConfig
+
+
+class SauceLabsVisual:
+    client: Client = None
+    build_id: str = None
+
+    def __init__(self):
+        self._create_client()
+
+    def _create_client(self):
+        region_url = Region.from_name(environ.get("SAUCE_REGION") or 'us-west-1').graphql_endpoint
+        username = environ.get("SAUCE_USERNAME")
+        access_key = environ.get("SAUCE_ACCESS_KEY")
+
+        if username is None or access_key is None:
+            raise Exception(
+                'Sauce Labs credentials not set. Please check that you set correctly your `user` '
+                'and `key`.'
+            )
+
+        transport = AIOHTTPTransport(
+            url=region_url,
+            auth=BasicAuth(environ.get("SAUCE_USERNAME"), environ.get("SAUCE_ACCESS_KEY")),
+        )
+        self.client = Client(transport=transport)
+
+    def get_client(self) -> Client:
+        return self.client
+
+    def create_build(
```

Comment:
should we consider converting those parameters into option object instead? Later we can add other options and managing them via parameters might be painful

Reply:
They're all named parameters so they can be passed in an order or completely omitted. We also get decent typing like this without forcing users to instantiate `TypedDict`s just to pass an option. But I'm open to changing it if the whole team feels otherwise.